test/scripts/products/h24-verify-selfmanaged-apicast-and-custom-policy/h24-verify-selfmanaged-apicast-and-custom-policy.go

```diff
@@ -0,0 +1,576 @@
+	routes := routev1.RouteList{}
+	err := client.List(context.TODO(), &routes, &k8sclient.ListOptions{
+		Namespace: threeScaleNamespace,
+	})
+	if err != nil {
+		log.Error("Error obtaining routes list in "+threeScaleNamespace+" namespace  ", err)
+		return err
+	}
+	for _, route := range routes.Items {
+		if strings.Contains(route.Spec.Host, "api-3scale-apicast-") {
+			err := client.Delete(ctx, &route)
+			if err != nil {
+				log.Error("error Delete Route", err)
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// Install 3scale APIcast gateway operator
+func installThreeScaleApicastGatewayOperator(ctx context.Context, client k8sclient.Client) error {
+	log.Info("Install 3scale APIcast gateway Operator")
+	targetNamespaces := []string{apicastNamespace}
+	og := &v1.OperatorGroup{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: apicastNamespace + "-",
+			Namespace:    apicastNamespace,
+			Generation:   1,
+			Annotations: map[string]string{
+				"olm.providedAPIs": "APIcast.v1alpha1.apps.3scale.net",
+			},
+		},
+		Spec: v1.OperatorGroupSpec{
+			TargetNamespaces: targetNamespaces,
+		},
+	}
+	err := client.Create(context.TODO(), og)
+	if err != nil && !k8serr.IsAlreadyExists(err) {
+		log.Error("Error create Operator Group in "+apicastNamespace+" namespace", err)
+		return err
+	}
+	subscription := &operatorsv1alpha1.Subscription{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "apicast-operator",
+			Namespace: apicastNamespace,
+			Labels: map[string]string{
+				"operators.coreos.com/apicast-operator.selfmanaged-apicast": "",
+			},
+			Generation: 1,
+		},
+		Spec: &operatorsv1alpha1.SubscriptionSpec{
+			Channel:                "threescale-" + threeScaleVersion,
+			InstallPlanApproval:    operatorsv1alpha1.ApprovalAutomatic,
+			Package:                "apicast-operator",
+			CatalogSource:          "redhat-operators",
+			CatalogSourceNamespace: "openshift-marketplace",
+			StartingCSV:            "apicast-operator." + apicastOperatorVersion,
+		},
+	}
+	err = client.Create(context.TODO(), subscription)
+	if err != nil {
+		log.Error("Error create Operator Subscription in "+apicastNamespace+" namespace", err)
+		return err
+	}
+	return nil
+}
+
+// Create a self-managed APIcast
+func createSelfManagedApicast(ctx context.Context, client k8sclient.Client) error {
+	log.Info("Create self-managed APIcast")
+	specConfigurationLoadMode := "boot"
+	//specImage := "image-registry.openshift-image-registry.svc:5000/selfmanaged-apicast/apicast-policy:example"
+	specReplicas := int64(1)
+	apicast := &appsv1alpha1.APIcast{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "example-apicast",
+			Namespace: apicastNamespace,
+			Labels: map[string]string{
+				"operators.coreos.com/apicast-operator.selfmanaged-apicast": "",
+			},
+			Generation: 2,
+			Annotations: map[string]string{
+				"apicast.apps.3scale.net/operator-version": "0.5.0",
```

Comment:
Do we anticipate this version to change in the future? If so can it be exposed so we don't need to update the test?

Reply:
Yes, We will handle it in https://issues.redhat.com/browse/MGDAPI-3037

Reply:
Done